src/main/java/cat/udl/eps/softarch/learnphysics/repository/LevelRepository.java

```diff
@@ -1,9 +1,17 @@
 package cat.udl.eps.softarch.learnphysics.repository;
 
 import cat.udl.eps.softarch.learnphysics.domain.Level;
+import cat.udl.eps.softarch.learnphysics.domain.Topic;
 import org.springframework.data.repository.PagingAndSortingRepository;
-import org.springframework.data.rest.core.annotation.RepositoryRestResource;
+import org.springframework.data.repository.query.Param;
+
+import java.util.Optional;
 
-@RepositoryRestResource
 public interface LevelRepository extends PagingAndSortingRepository<Level, Integer> {
+    Optional<Level> findById(@Param("levelId") String id);
+    Level findByName(@Param("name") String name);
+    Level findByDescription(@Param("description") String description);
+    Level findByTopic(@Param("topics") Topic topic);
```

Comment:
The name of the property is `topics` so this function should be `findByTopics`. However, you don't actually need this functions right now.

Reply:
This is still causing trouble